### PR TITLE
NAS-113464 / None / Added ix-table-extended

### PR DESCRIPTION
**Pull request by RehanY147**
The `ix-table-extended` approach needs to be reviewed. Plus typical code review. The NTP Servers List under General Settings is used as an example.

### CODE REVIEW

**Review by denysbutenko — APPROVED** [review]
Looks good to me. Perhaps it can replace `app-table` on the Sharing and Data Protection pages

File: src/app/modules/ix-tables/components/ix-table-extended/ix-table-extended.component.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    switch (this.status) {
+      case IxTableStatus.Loading:
+        return this.loadingConf;
+        break;
+      case IxTableStatus.Error:
+        return this.errorConf;
+        break;
+      default:
+        return this.emptyConf;
+        break;
```
Comment (AlexKarpov98) [inline src/app/modules/ix-tables/components/ix-table-extended/ix-table-extended.component.ts:59]: May remove break lines.

**Review by AlexKarpov98 — APPROVED** [review]
LGTM! 👍
Screenshot_15
Screenshot_16

**Review by undsoft — CHANGES_REQUESTED** [review]
I cannot say that I like the approach. 
One of the major design goals with entity-table was to have a set of components that can be connected together to build the table. As in not do what entity-table was doing by wrapping logic inside of component. This goes back on this goal.
For example, I don't see why title or buttons should be part of the `table` component.
The table tag is now hidden from a developer.

What I really wanted by this ticket is to reduce boilerplate by taking all of these emptyConf, loadingConf, errorConf and extracting that part in a way that can be connected to ix-table.

**Pull request closed without merging**